src/main/java/com/personal/todolist/controller/RestExceptionInterceptor.java

```diff
@@ -0,0 +1,17 @@
+package com.personal.todolist.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class RestExceptionInterceptor extends ResponseEntityExceptionHandler {
+    @ExceptionHandler(value = { IllegalArgumentException.class, IllegalStateException.class })
+    public ResponseEntity<Object> handleConflict(RuntimeException ex, WebRequest request) {
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).contentType(MediaType.APPLICATION_JSON).body(ex);
```

Comment:
Wrzuciłeś cały obiekt "ex" do body.
Generalnie nie jest to dobra metodyka, bo obiekt ten zawiera "stack trace" - informacje ze stosu w którym miejscu programu poszedł wyjątek. Zawiera on nazwy paczek i klas, które używa projekt. Jest to potencjalnie niebezpieczna informacja, z nią ktoś może próbować wstrzyknąć inny obiekt. Zwróć proszę tutaj tylko wiadomość z wyjątku:
`ex.getMessage()`